src/Search/Search.Tests/Tests/CustomAnalyzerTests.cs

```diff
@@ -650,7 +650,7 @@ private static void AssertIndexContainsAllAnalysisComponents(params Index[] inde
                 select new { Type = typeGroup.Key, Count = typeGroup.Count() };
 
             Dictionary<Type, int> instanceCountMap = analysisTypeCounts.ToDictionary(tc => tc.Type, tc => tc.Count);
-            IEnumerable<Type> allAnalysisComponentTypes = typeof(Index).GetTypeInfo().Assembly.ExportedTypes.Where(IsAnalysisComponentType);
+            IEnumerable<Type> allAnalysisComponentTypes = typeof(Index).GetTypeInfo().Assembly.ExportedTypes.Where((Type type) => IsAnalysisComponentType(type) && !IsDeprecatedType(type));
```

Comment:
You don't need to type lambda parameters, and if there's only one, you don't need parentheses either.